src/plugins/enzymes/src/FindEnzymesDialog.cpp

```diff
@@ -336,6 +363,27 @@ void EnzymesSelectorWidget::sl_filterTextChanged(const QString& filterText) {
     }
 }
 
+void EnzymesSelectorWidget::sl_findSingleEnzymeTaskStateChanged() {
+    auto t = qobject_cast<FindSingleEnzymeTask*>(sender());
+    SAFE_POINT(t != nullptr, L10N::nullPointerError("FindSingleEnzymeTask"), );
+
+    CHECK(!t->hasError(), );
+    CHECK(t->getState() == Task::State_Finished, );
+
+    auto taskEnzyme = t->getEnzyme();
+    auto taskItem = getEnzymeTreeItemByEnzymeData(taskEnzyme);
+    CHECK(taskItem != nullptr, );
+
+    int size = t->getResults().size();
+    bool maxResultsFound = t->isCanceled() && size == EnzymeTreeItem::MAXIMUM_ENZYMES_NUMBER;
```

Comment:
If the task is canceled you can't trust the result: I can cancel it via Task manager.
The task should use max-enzymes limit and finish with no errors & not canceled state.

Reply:
You cannot cancel task because of the modal dialog and it could be not easy to rewrite `FindSingleEnzymeTask` as you want. Not sure if it's worth it

Reply:
The important part here is that you can't trust result of a canceled task. The cancellation may happen by different reasons.

Reply:
OK